HADOOP-18830. Cut S3 Select

steveloughran opened the pull request:
Cut out S3 Select
* leave public/unstable constants alone
* s3guard tool will fail with error
* s3afs. path capability will fail
* openFile() will fail with specific error
* s3 select doc updated

Everything else cut, including tests.

New test: ITestSelectUnsupported

verifies new failure handling above

### For code changes:

- [X] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [x] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

steveloughran committed:
HADOOP-18830-cut-s3-select

Cut out S3 Select
* leave public/unstable constants alone
* s3guard tool will fail with error
* s3afs. path capability will fail
* openFile() will fail with specific error
* s3 select doc updated
* Cut eventstream jar
* New test: ITestSelectUnsupported verifies new failure
  handling above

Everything else cut, including tests.

Change-Id: Iff8c1e6562ed2d47abefe2cfef2909c00c78a55d

steveloughran commented:
Given I wrote this it's kind of sad, but openFile() will outlive it as it has many more uses.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 7 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  48m 20s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 41s |  |  trunk passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |   0m 32s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 40s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 25s |  |  trunk passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 32s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  37m 48s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |   0m 25s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  |  hadoop-tools/hadoop-aws: The patch generated 0 new + 4 unchanged - 8 fixed = 4 total (was 12)  |
| +1 :green_heart: |  mvnsite  |   0m 31s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 23s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   1m  6s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  38m 24s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 43s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  |  The patch does not generate ASF License warnings.  |
|  |   | 141m 15s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux 1bb711adcce1 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / dc6c3c358853b02a8d30a12490a08e353dec9a75 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/1/testReport/ |
| Max. process+thread count | 529 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
Tested s3 london `Dparallel-tests -DtestsThreadCount=10 -Dscale -Dprefetch`

steveloughran commented:
with this PR
* all s3 select code is stripped
* all imports of its classes are gone
* docs say this
* path capability returns false
* openFile with select options raise UnsupportedOperation + test
* s3guard select command fails, with new test

Retained
1. `SelectConstants` in case used externally
2. `org.apache.hadoop.fs.s3a.select.BlockingEnumeration` in case its useful in future

I have a concern about retention #2; it uses the unshaded package `org.reactivestreams` hidden away in bundle.jar. I understand why it is unshaded -it's a public API- but its not
good to build against someone's classes in an unrelated jar...wildfly springs to mind as an example

we may want to cut it -we can always add a ref in the package docs to say its' been cut and which git commit to go to so as to find it.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 9 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m  1s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  35m 54s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 33s |  |  trunk passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 58s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 45s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  19m 35s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   8m 54s |  |  trunk passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 27s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |  32m 16s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  69m 34s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  34m 16s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 37s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 49s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  16m 49s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m 40s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/2/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 1 new + 5 unchanged - 8 fixed = 6 total (was 13)  |
| +1 :green_heart: |  mvnsite  |  15m  0s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |   8m 38s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 31s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |  32m 42s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  68m 38s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 768m 27s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 1150m 42s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint shellcheck shelldocs |
| uname | Linux e6966d879857 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / ab2ef9c77b6060921cd9ded49a24c0ad0a44e561 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/2/testReport/ |
| Max. process+thread count | 2608 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 49s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 9 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  16m 14s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  36m  4s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 25s |  |  trunk passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 56s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 46s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  20m 46s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   8m 59s |  |  trunk passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 25s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 18s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  69m  5s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  35m  1s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 10s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m  4s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  17m  4s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 29s |  |  root: The patch generated 0 new + 5 unchanged - 8 fixed = 5 total (was 13)  |
| +1 :green_heart: |  mvnsite  |  15m  4s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |   8m 47s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 26s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 17s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  69m  3s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 764m  2s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/3/artifact/out/patch-unit-root.txt) |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   | 1151m 43s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.datanode.TestDirectoryScanner |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint shellcheck shelldocs |
| uname | Linux 405b2c270aef 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 616e55adb75d5accae95493fe95a835935168043 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/3/testReport/ |
| Max. process+thread count | 2741 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

ahmarsuhail commented on hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/index.md:
```diff
@@ -35,7 +35,6 @@ full details.
 
 * [Encryption](./encryption.html)
 * [Performance](./performance.html)
-* [S3Guard](./s3guard.html)
```
why are we removing this? think the S3Guard page is still there

steveloughran replied:
think i moved it

ahmarsuhail reviewed (commented):
Sad to see this go too..it complicated the SDK V2 work and we had to add in a lot of code to get this work.

Looks good, few things:
* Remove the CsvFile class?
* In S3Utils we have SC_405_METHOD_NOT_ALLOWED, comment says "seen on S3 Select" . maybe we don't need that anymore
* Couple of open todos in `ITestSelectUnsupported`

ahmarsuhail commented:
And agree with your comment, should cut `org.apache.hadoop.fs.s3a.select.BlockingEnumeration` and add the commit to the package-info.

steveloughran commented:
keeping but moving csv file. it's actually very useful in places; in cloudstore we use it in places like "bandwidth" to track block upload times. so it may be useful in future...leaving it in test/ means there's no commitment to maintain

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 50s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 13 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  16m  1s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  36m  4s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 17s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 41s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 40s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  20m 14s |  |  trunk passed  |
| -1 :x: |  javadoc  |   2m 44s | [/branch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/4/artifact/out/branch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt) |  root in trunk failed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  javadoc  |   7m 28s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 18s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  68m 20s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  34m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 45s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 45s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 47s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  16m 47s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 33s |  |  root: The patch generated 0 new + 5 unchanged - 8 fixed = 5 total (was 13)  |
| +1 :green_heart: |  mvnsite  |  15m  4s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   2m 34s | [/patch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/4/artifact/out/patch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  javadoc  |   7m 30s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 18s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  70m 58s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 451m 11s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/4/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 26s |  |  The patch does not generate ASF License warnings.  |
|  |   | 824m 54s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.datanode.TestDirectoryScanner |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint shellcheck shelldocs |
| uname | Linux 5d804cf3cb82 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a4346dd7dec4acdcee745453cd4ce49531cb2282 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/4/testReport/ |
| Max. process+thread count | 2441 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/4/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 46s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 13 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 42s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  38m 55s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  19m 10s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  17m 18s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 43s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  20m 44s |  |  trunk passed  |
| -1 :x: |  javadoc  |   2m 53s | [/branch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6144/1/artifact/out/branch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt) |  root in trunk failed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  javadoc  |   8m  3s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 18s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  71m 22s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  35m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 42s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  21m 48s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  21m 48s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 57s |  |  root: The patch generated 0 new + 5 unchanged - 8 fixed = 5 total (was 13)  |
| +1 :green_heart: |  mvnsite  |  15m 54s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   2m 36s | [/patch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6144/1/artifact/out/patch-javadoc-root-jdkUbuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  javadoc  |   7m 37s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 17s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  70m 17s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 449m  3s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6144/1/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m  7s |  |  The patch does not generate ASF License warnings.  |
|  |   | 838m 16s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.datanode.TestDirectoryScanner |
|   | hadoop.yarn.server.timeline.webapp.TestTimelineWebServices |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6144/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint shellcheck shelldocs |
| uname | Linux ab1efc20e385 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a4346dd7dec4acdcee745453cd4ce49531cb2282 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6144/1/testReport/ |
| Max. process+thread count | 2742 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6144/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   9m  5s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 13 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 22s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  20m 24s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  11m 28s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   9m 35s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  13m  0s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   5m 47s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   4m 58s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 18s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  37m  7s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  17m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 59s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   8m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 30s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |   8m 30s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 17s |  |  root: The patch generated 0 new + 5 unchanged - 8 fixed = 5 total (was 13)  |
| +1 :green_heart: |  mvnsite  |   8m 39s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |   5m 38s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   4m 56s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 17s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  36m 26s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 648m  0s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 19s |  |  The patch does not generate ASF License warnings.  |
|  |   | 880m  4s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint shellcheck shelldocs |
| uname | Linux 1086dca886f0 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 88f202d35a049030876dabf54ac2f252b85ee689 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/5/testReport/ |
| Max. process+thread count | 3372 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/5/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
latest change is just a rebase; not retested

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  18m 35s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 13 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 52s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  35m 18s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 23s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  18m  1s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   4m 50s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  19m 37s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   8m 46s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 31s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 18s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  67m 23s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  33m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 29s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 29s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 35s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  16m 35s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 36s |  |  root: The patch generated 0 new + 6 unchanged - 8 fixed = 6 total (was 14)  |
| +1 :green_heart: |  mvnsite  |  15m 18s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |   9m  5s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 37s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 19s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  67m 25s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 775m 15s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/7/artifact/out/patch-unit-root.txt) |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   | 1173m 30s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.fs.s3a.prefetch.TestS3ACachingBlockManager |
|   | hadoop.hdfs.server.datanode.TestDirectoryScanner |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint shellcheck shelldocs |
| uname | Linux 6cfc617a538b 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / eadbd958d3c04bcb6d4e388e41a863998cb027fe |
| Default Java | Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/7/testReport/ |
| Max. process+thread count | 3001 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/7/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
rebased pr with retest. failures unrelated; the signing one has an active pr to fix, the committer one looks like my config is at fault (bucket overrides not being cut)
```
[ERROR] Failures: 
[ERROR]   ITestS3ACommitterFactory.testEverything:115->testInvalidFileBinding:165 Expected a org.apache.hadoop.fs.s3a.commit.PathCommitException to be thrown, but got the result: : FileOutputCommitter{PathOutputCommitter{context=TaskAttemptContextImpl{JobContextImpl{jobId=job_202401050108_0001}; taskId=attempt_202401050108_0001_m_000000_0, status=''}; org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter@61fa8914}; outputPath=s3a://stevel--usw2-az1--x-s3/fork-0001/test/testEverything, workPath=s3a://stevel--usw2-az1--x-s3/fork-0001/test/testEverything/_temporary/1/_temporary/attempt_202401050108_0001_m_000000_0, algorithmVersion=1, skipCleanup=false, ignoreCleanupFailures=false}
[ERROR] Errors: 
[ERROR]   ITestCustomSigner.testCustomSignerAndInitializer:135->runStoreOperationsAndVerify:155->lambda$runStoreOperationsAndVerify$0:160 » AWSBadRequest
[ERROR]   ITestCustomSigner.testCustomSignerAndInitializer:135->runStoreOperationsAndVerify:155->lambda$runStoreOperationsAndVerify$0:160 » AWSBadRequest
[INFO] 
```

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 53s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 13 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 38s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  30m 43s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  16m 11s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  14m 51s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   4m 13s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 33s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 15s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 22s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 42s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  33m  7s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 43s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 38s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  15m 38s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 47s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  14m 47s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m  2s |  |  root: The patch generated 0 new + 6 unchanged - 8 fixed = 6 total (was 14)  |
| +1 :green_heart: |  mvnsite  |   1m 29s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 10s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 21s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 36s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  32m 59s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 32s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 55s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 59s |  |  The patch does not generate ASF License warnings.  |
|  |   | 206m  3s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint |
| uname | Linux e7bd09825574 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a254800c511f9a3cd3f78e399e6bb834c69132fc |
| Default Java | Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/8/testReport/ |
| Max. process+thread count | 750 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/8/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
@ahmarsuhail @mukund-thakur can I get this in? it'll help on test time...

ahmarsuhail commented on hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/select/package-info.java:
```diff
@@ -17,7 +17,12 @@
 /**
- * Support for S3 Select.
+ * Was the location for support for S3 Select.
+ * Now removed apart from some constants.f
```
nit: typo

ahmarsuhail commented on hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/select/ITestSelectUnsupported.java:
```diff
@@ -0,0 +1,100 @@
+      " WHERE s._1 = 'foo'";
+
+  /**
+   * A {@code .must(SELECT_SQL, _)} option MSUT not fail the build.
```
nit: typo in MUST
and this isn't clear to me, .must _should_ fail and throw Unsupported Exception right?

steveloughran replied:
correct. updated this and other javadocs, named methods appropriately

ahmarsuhail reviewed (approved):
+1, LGTM. just a couple of javadoc nits.

steveloughran committed:
HADOOP-18830. S3 select. final review comments

+update doc with explicit consequences of removal,
including the "hadoop s3guard select" command

Change-Id: Ica3ffb33d803f8a03e0d733447e4f026d547f8c1

ahmarsuhail reviewed (approved):
+1, LGTM.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  docker  |  11m 36s |  |  Docker failed to build run-specific yetus/hadoop:tp-16309}.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/9/console |
| versions | git=2.34.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
thanks. not sure what is up with docker. will rebase and force push with test run

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  10m 21s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 13 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 26s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  35m 24s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 58s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 34s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   4m 38s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 29s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 11s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 18s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 40s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  38m 39s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 17s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 17s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 26s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  16m 26s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 37s |  |  root: The patch generated 0 new + 6 unchanged - 8 fixed = 6 total (was 14)  |
| +1 :green_heart: |  mvnsite  |   1m 26s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m  9s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 17s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 33s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  39m  0s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 33s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  1s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 57s |  |  The patch does not generate ASF License warnings.  |
|  |   | 239m 12s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/10/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint |
| uname | Linux cc9c48585b05 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 736a3ccf194bb6a3082b65e85f121d1b9302f991 |
| Default Java | Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/10/testReport/ |
| Max. process+thread count | 578 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/10/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran committed:
HADOOP-18830. Final doc fixup

Review in docs/core-siten references to landsat and CSV;
update as appropriate.

This includes
* A review of use of .endpoint vs .endpoint.region
  and a move to fs.s3a.endpoint.region as much as possible.
* reorg of some sections
* fixup of a name= headers.

Change-Id: I5e07cdf153cacf0ce1ee6673d3a094c2d5eaf5a2

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 47s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 14 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 49s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  34m 56s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 11s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 38s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   4m 39s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 29s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 10s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 16s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 39s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  38m 14s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   1m  0s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 15s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 18s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  16m 18s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/11/artifact/out/blanks-eol.txt) |  The patch has 1 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  checkstyle  |   4m 36s |  |  root: The patch generated 0 new + 6 unchanged - 8 fixed = 6 total (was 14)  |
| +1 :green_heart: |  mvnsite  |   1m 25s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m  8s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 18s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +0 :ok: |  spotbugs  |   0m 34s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  39m 40s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 34s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  0s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 59s |  |  The patch does not generate ASF License warnings.  |
|  |   | 230m 25s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/11/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6144 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle markdownlint |
| uname | Linux 1664d8f960c5 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 034865ddba406e47530f1bceb19c824081bf2ebe |
| Default Java | Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/11/testReport/ |
| Max. process+thread count | 528 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6144/11/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.
